src/app/AppContainer.js

```diff
@@ -2,55 +2,109 @@
  * @format */
 
 import { connect } from 'react-redux';
-import {
-	updateBlockAttributes,
-	focusBlockAction,
-	moveBlockUpAction,
-	moveBlockDownAction,
-	deleteBlockAction,
-	createBlockAction,
-	parseBlocksAction,
-	serializeToNativeAction,
-	mergeBlocksAction,
-} from '../store/actions';
+import { isEqual } from 'lodash';
+
 import MainApp from './MainApp';
+import { parse, serialize } from '@wordpress/blocks';
+import { withDispatch, withSelect } from '@wordpress/data';
+import { compose } from '@wordpress/compose';
+import RNReactNativeGutenbergBridge from 'react-native-gutenberg-bridge';
+
+const mapStateToProps = ( state, ownProps ) => {
+	let blocks = state.blocks;
+	let refresh = !! ownProps.refresh;
```

Comment:
It's given as props to BlockManager a few lines below https://github.com/wordpress-mobile/gutenberg-mobile/pull/226/files#diff-0b657b049f5ed3fc0862ff8bf5c9e4d4R29

Then to `FlatList` here https://github.com/wordpress-mobile/gutenberg-mobile/pull/226/files#diff-2d50daf729e068ecebcce113d1ce7f9aR254

Reply:
Hmm, let me see I understand this. `mapStateToProps` will be called for every component down the line so, `BlockManager` will get the `refresh` prop from the `mapStateToProps()` call on the component _above_ `BlockManager`?

Plus, `ownProps.refresh` will be undefined for `AppContainer`?

Reply:
Not really for every component, `connect` is just an HOC that will re-render the inside component (`MainApp` in this case) on state change (meaning it will bind listeners on `this.props.store` given by the provider in `App`).
So every properties returned in `mapStateToProps` are added to the props of `MainApp` and as you can see in `MainApp.js` https://github.com/wordpress-mobile/gutenberg-mobile/blob/master/src/app/MainApp.js#L33 this component passes all its props down to `BlockManager`.
In this case `refresh` is a hack for `FlatList`, we're just flipping it's value whenever we detect a change in `blocks`. Later we'll probably want to remove it and rely on blocks only.

Reply:
> So every properties returned in mapStateToProps are added to the props of MainApp and as you can see in MainApp.js

Cool, that's what I've understood so far, thanks. So, I'm back to the original question: where is the value of `ownProps.refresh` set? My impression is that it's always `undefined` since `AppContainer`'s props never get augmented with `refresh`. Right?

Reply:
You are right, it actually isn't set.
Yeah we should just be using `false` instead.